### PR TITLE
dm: add error binlog filter description

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/15309
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-binlog-event-filter.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ summary: Learn how to use the binlog event filter feature of DM.
 
 # TiDB Data Migration Binlog Event Filter
 
-TiDB Data Migration (DM) provides the binlog event filter feature to filter out or only receive specified types of binlog events for some schemas or tables. For example, you can filter out all `TRUNCATE TABLE` or `INSERT` events. The binlog event filter feature is more fine-grained than the [block and allow lists](/dm/dm-block-allow-table-lists.md) feature.
+TiDB Data Migration (DM) provides the binlog event filter feature to filter out, report errors, or only receive specified types of binlog events for some schemas or tables. For example, you can filter out all `TRUNCATE TABLE` or `INSERT` events. The binlog event filter feature is more fine-grained than the [block and allow lists](/dm/dm-block-allow-table-lists.md) feature.
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:8]:
```suggestion
TiDB Data Migration (DM) provides the binlog event filter feature to filter out, block and report errors, or only receive specified types of binlog events for some schemas or tables. For example, you can filter out all `TRUNCATE TABLE` or `INSERT` events. The binlog event filter feature is more fine-grained than the [block and allow lists](/dm/dm-block-allow-table-lists.md) feature.
```

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: dm/dm-binlog-event-filter.md [hunk 2]
```diff
@@ -148,3 +175,16 @@ filters:
     sql-pattern: ["ALTER\\s+TABLE[\\s\\S]*ADD\\s+PARTITION", "ALTER\\s+TABLE[\\s\\S]*DROP\\s+PARTITION"]
     action: Ignore
 ```
+
+### Report errors on some DDL statements
+
+If you need to report errors on DDL statements for some upstream operations, use the following settings:
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:181]:
```suggestion
If you need to block and report errors on DDL statements generated by some upstream operations before the DM replicates data to TiDB, you can use the following settings:
```

File: dm/dm-binlog-event-filter.md [hunk 3]
```diff
@@ -40,33 +40,60 @@ When you use the wildcard for matching schemas and tables, note the following:
...
-    | `drop table`      | DDL  | The `DROP TABLE` DDL event              |
-    | `truncate table`  | DDL  | The `TRUNCATE TABLE` DDL event          |
-    | `rename table`    | DDL  | The `RENAME TABLE` DDL event            |
-    | `drop index`      | DDL  | The `DROP INDEX` DDL event           |
+    | `drop table`      | incompatible DDL  | The `DROP TABLE` DDL event              |
+    | `truncate table`  | incompatible DDL  | The `TRUNCATE TABLE` DDL event          |
+    | `rename table`    | incompatible DDL  | The `RENAME TABLE` DDL event            |
+    | `drop index`      | incompatible DDL  | The `DROP INDEX` DDL event           |
     | `alter table`     | DDL  | The `ALTER TABLE` DDL event           |
+    | `value range decrease` | incompatible DDL  | A DDL statement that makes a column field smaller, such as the `ALTER TABLE MODIFY COLUMN` statement that changes `VARCHAR(20)` to `VARCHAR(10)`  |
```
Comment (Frank945946) [inline dm/dm-binlog-event-filter.md:59]:
```suggestion
    | `value range decrease` | incompatible DDL  | A DDL statement that makes a column field value range smaller, such as the `ALTER TABLE MODIFY COLUMN` statement that changes `VARCHAR(20)` to `VARCHAR(10)`  |
```

File: dm/dm-binlog-event-filter.md [hunk 4]
```diff
@@ -148,3 +175,16 @@ filters:
     sql-pattern: ["ALTER\\s+TABLE[\\s\\S]*ADD\\s+PARTITION", "ALTER\\s+TABLE[\\s\\S]*DROP\\s+PARTITION"]
     action: Ignore
 ```
+
+### Report errors on some DDL statements
+
+If you need to block and report errors on DDL statements generated by some upstream operations before the DM replicates data to TiDB, you can use the following settings:
```
Comment (Frank945946) [inline dm/dm-binlog-event-filter.md:181]:
```suggestion
If you need to block and report errors on DDL statements generated by some upstream operations before the DM replicates them to TiDB, you can use the following settings:
```

**Review by Frank945946 — APPROVED** [review]

File: dm/dm-binlog-event-filter.md [hunk 5]
```diff
@@ -40,33 +40,60 @@ When you use the wildcard for matching schemas and tables, note the following:
...
-    | `drop table`      | DDL  | The `DROP TABLE` DDL event              |
-    | `truncate table`  | DDL  | The `TRUNCATE TABLE` DDL event          |
-    | `rename table`    | DDL  | The `RENAME TABLE` DDL event            |
-    | `drop index`      | DDL  | The `DROP INDEX` DDL event           |
+    | `drop table`      | incompatible DDL  | The `DROP TABLE` DDL event              |
+    | `truncate table`  | incompatible DDL  | The `TRUNCATE TABLE` DDL event          |
+    | `rename table`    | incompatible DDL  | The `RENAME TABLE` DDL event            |
+    | `drop index`      | incompatible DDL  | The `DROP INDEX` DDL event           |
     | `alter table`     | DDL  | The `ALTER TABLE` DDL event           |
+    | `value range decrease` | incompatible DDL  | A DDL statement that makes the value range of a column field smaller, such as the `ALTER TABLE MODIFY COLUMN` statement that changes `VARCHAR(20)` to `VARCHAR(10)`  |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:59]:
```suggestion
    | `value range decrease` | incompatible DDL  | A DDL statement that decreases the value range of a column field, such as the `ALTER TABLE MODIFY COLUMN` statement that changes `VARCHAR(20)` to `VARCHAR(10)`  |
```

File: dm/dm-binlog-event-filter.md [hunk 6]
```diff
@@ -40,33 +40,60 @@ When you use the wildcard for matching schemas and tables, note the following:
...
+    | `truncate table partition` | incompatible DDL  | A DDL statement that truncates a partition in a table, such as the `ALTER TABLE TRUNCATE PARTITION` statement |
+    | `drop primary key` | incompatible DDL  | A DDL statement that drops the primary key, such as the `ALTER TABLE DROP PRIMARY KEY` statement |
+    | `drop unqiue key` | incompatible DDL  | A DDL statement that drops a unique key, such as the `ALTER TABLE DROP UNIQUE KEY` statement |
+    | `modify default value` | incompatible DDL  | A DDL statement that modifies a column's default value, such as the `ALTER TABLE CHANGE DEFAULT` statement |
+    | `modify constraint` | incompatible DDL  | A DDL statement that modifies the constraint, such as the `ALTER TABLE ADD CONSTRAINT` statement |
+    | `modify columns order` | incompatible DDL  | A DDL statement that modifies the order of the columns, such as the `ALTER TABLE CHANGE AFTER` statement |
+    | `modify charset` | incompatible DDL  | A DDL statement that modifies a column's charset, such as the `ALTER TABLE MODIFY CHARSET` statement |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:72]:
```suggestion
    | `modify charset` | incompatible DDL  | A DDL statement that modifies the charset of a column, such as the `ALTER TABLE MODIFY CHARSET` statement |
```

File: dm/dm-binlog-event-filter.md [hunk 7]
```diff
@@ -40,33 +40,60 @@ When you use the wildcard for matching schemas and tables, note the following:
...
+    | `modify collation` | incompatible DDL  | A DDL statement that modifies a column collation, such as the `ALTER TABLE MODIFY COLLATE` statement |
+    | `remove auto increment` | incompatible DDL  | A DDL statement that removes a auto-incremental key |
+    | `modify storage engine` | incompatible DDL  | A DDL statement that modifies the table storage engine, such as the `ALTER TABLE ENGINE = MyISAM` statement |
+    | `reorganize table partiton` | incompatible DDL  | A DDL statement that reorganizes the table partiton, such as the `ALTER TABLE REORGANIZE PARTITION` statement |
+    | `rebuild table partiton` | incompatible DDL  | A DDL statement that rebuilds the table partiton, such as the `ALTER TABLE REBUILD PARTITION` statement |
+    | `exchange table partiton` | incompatible DDL  | A DDL statement that exchanges the table partiton, such as the `ALTER TABLE EXCHANGE PARTITION` statement |
+    | `coalesce table partiton` | incompatible DDL  | A DDL statement that coalesces the table partiton, such as the `ALTER COALESCE PARTITION` statement |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:79]:
```suggestion
    | `reorganize table partition` | incompatible DDL  | A DDL statement that reorganizes the table partition, such as the `ALTER TABLE REORGANIZE PARTITION` statement |
    | `rebuild table partition` | incompatible DDL  | A DDL statement that rebuilds the table partition, such as the `ALTER TABLE REBUILD PARTITION` statement |
    | `exchange table partition` | incompatible DDL  | A DDL statement that exchanges the table partition, such as the `ALTER TABLE EXCHANGE PARTITION` statement |
    | `coalesce table partition` | incompatible DDL  | A DDL statement that coalesces the table partition, such as the `ALTER COALESCE PARTITION` statement |
```

File: dm/dm-binlog-event-filter.md [hunk 8]
```diff
@@ -40,33 +40,60 @@ When you use the wildcard for matching schemas and tables, note the following:
...
-    | `truncate table`  | DDL  | The `TRUNCATE TABLE` DDL event          |
-    | `rename table`    | DDL  | The `RENAME TABLE` DDL event            |
-    | `drop index`      | DDL  | The `DROP INDEX` DDL event           |
+    | `drop table`      | incompatible DDL  | The `DROP TABLE` DDL event              |
+    | `truncate table`  | incompatible DDL  | The `TRUNCATE TABLE` DDL event          |
+    | `rename table`    | incompatible DDL  | The `RENAME TABLE` DDL event            |
+    | `drop index`      | incompatible DDL  | The `DROP INDEX` DDL event           |
     | `alter table`     | DDL  | The `ALTER TABLE` DDL event           |
+    | `value range decrease` | incompatible DDL  | A DDL statement that makes the value range of a column field smaller, such as the `ALTER TABLE MODIFY COLUMN` statement that changes `VARCHAR(20)` to `VARCHAR(10)`  |
+    | `precision decrease` | incompatible DDL  | A DDL statement that reduces the precision of a column field, such as the `ALTER TABLE MODIFY COLUMN` statement that changes `Decimal(10, 2)` to `Decimal(10, 1)`  |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:60]:
```suggestion
    | `precision decrease` | incompatible DDL  | A DDL statement that decreases the precision of a column field, such as the `ALTER TABLE MODIFY COLUMN` statement that changes `Decimal(10, 2)` to `Decimal(10, 1)`  |
```

File: dm/dm-binlog-event-filter.md [hunk 9]
```diff
@@ -40,33 +40,60 @@ When you use the wildcard for matching schemas and tables, note the following:
     | `all`             |      | Includes all the events below |
     | `all dml`         |      | Includes all DML events below |
     | `all ddl`         |      | Includes all DDL events below |
+    | `incompatible ddl changes` |      | Includes all incompatible DDL events     |
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:43]:
```suggestion
    | `incompatible ddl changes` |      | Includes all incompatible DDL events, i.e. DDLs that can cause data loss     |
```

File: dm/dm-binlog-event-filter.md [hunk 10]
```diff
@@ -40,33 +40,60 @@ When you use the wildcard for matching schemas and tables, note the following:
...
+    | `modify constraint` | incompatible DDL  | A DDL statement that modifies the constraint, such as the `ALTER TABLE ADD CONSTRAINT` statement |
+    | `modify columns order` | incompatible DDL  | A DDL statement that modifies the order of the columns, such as the `ALTER TABLE CHANGE AFTER` statement |
+    | `modify charset` | incompatible DDL  | A DDL statement that modifies the charset of a column, such as the `ALTER TABLE MODIFY CHARSET` statement |
+    | `modify collation` | incompatible DDL  | A DDL statement that modifies a column collation, such as the `ALTER TABLE MODIFY COLLATE` statement |
+    | `remove auto increment` | incompatible DDL  | A DDL statement that removes a auto-incremental key |
+    | `modify storage engine` | incompatible DDL  | A DDL statement that modifies the table storage engine, such as the `ALTER TABLE ENGINE = MyISAM` statement |
+    | `reorganize table partition` | incompatible DDL  | A DDL statement that reorganizes the table partition, such as the `ALTER TABLE REORGANIZE PARTITION` statement |
```
Comment (qiancai) [inline dm/dm-binlog-event-filter.md:76]:
```suggestion
    | `reorganize table partition` | incompatible DDL  | A DDL statement that reorganizes partitions in a table, such as the `ALTER TABLE REORGANIZE PARTITION` statement |
```

File: dm/dm-binlog-event-filter.md [hunk 11]
```diff
@@ -40,33 +40,60 @@ When you use the wildcard for matching schemas and tables, note the following:
...
+    | `rebuild table partition` | incompatible DDL  | A DDL statement that rebuilds the table partition, such as the `ALTER TABLE REBUILD PARTITION` statement |
+    | `exchange table partition` | incompatible DDL  | A DDL statement that exchanges the table partition, such as the `ALTER TABLE EXCHANGE PARTITION` statement |
+    | `coalesce table partition` | incompatible DDL  | A DDL statement that coalesces the table partition, such as the `ALTER COALESCE PARTITION` statement |
```
Comment (qiancai) [inline dm/dm-binlog-event-filter.md:79]:
```suggestion
    | `coalesce table partition` | incompatible DDL  | A DDL statement that decreases the number of partitions in a table, such as the `ALTER COALESCE PARTITION` statement |
```

File: dm/dm-binlog-event-filter.md [hunk 12]
```diff
@@ -40,33 +40,60 @@ When you use the wildcard for matching schemas and tables, note the following:
...
+    | `precision decrease` | incompatible DDL  | A DDL statement that decreases the precision of a column field, such as the `ALTER TABLE MODIFY COLUMN` statement that changes `Decimal(10, 2)` to `Decimal(10, 1)`  |
+    | `modify column` | incompatible DDL  | A DDL statement that changes the type of a column field, such as the `ALTER TABLE MODIFY COLUMN` statement that changes `INT` to `VARCHAR` |
+    | `rename column` | incompatible DDL  | A DDL statement that changes the name of a column, such as the `ALTER TABLE RENAME COLUMN` statement|
+    | `rename index` | incompatible DDL  | A DDL statement that changes the index name, such as the `ALTER TABLE RENAME INDEX` statement |
+    | `drop column` | incompatible DDL  | A DDL statement that drops a column from a table, such as the `ALTER TABLE DROP COLUMN` statement |
+    | `drop index` | incompatible DDL  | A DDL statement that drops an index in a table, such as the `ALTER TABLE DROP INDEX` statement |
+    | `truncate table partition` | incompatible DDL  | A DDL statement that truncates a partition in a table, such as the `ALTER TABLE TRUNCATE PARTITION` statement |
```
Comment (qiancai) [inline dm/dm-binlog-event-filter.md:66]:
```suggestion
    | `truncate table partition` | incompatible DDL  | A DDL statement that removes all data from a specified partition, such as the `ALTER TABLE TRUNCATE PARTITION` statement |
```

File: dm/dm-binlog-event-filter.md [hunk 13]
```diff
@@ -40,33 +40,60 @@ When you use the wildcard for matching schemas and tables, note the following:
...
+    | `modify charset` | incompatible DDL  | A DDL statement that modifies the charset of a column, such as the `ALTER TABLE MODIFY CHARSET` statement |
+    | `modify collation` | incompatible DDL  | A DDL statement that modifies a column collation, such as the `ALTER TABLE MODIFY COLLATE` statement |
+    | `remove auto increment` | incompatible DDL  | A DDL statement that removes a auto-incremental key |
+    | `modify storage engine` | incompatible DDL  | A DDL statement that modifies the table storage engine, such as the `ALTER TABLE ENGINE = MyISAM` statement |
+    | `reorganize table partition` | incompatible DDL  | A DDL statement that reorganizes the table partition, such as the `ALTER TABLE REORGANIZE PARTITION` statement |
+    | `rebuild table partition` | incompatible DDL  | A DDL statement that rebuilds the table partition, such as the `ALTER TABLE REBUILD PARTITION` statement |
+    | `exchange table partition` | incompatible DDL  | A DDL statement that exchanges the table partition, such as the `ALTER TABLE EXCHANGE PARTITION` statement |
```
Comment (qiancai) [inline dm/dm-binlog-event-filter.md:78]:
```suggestion
    | `exchange table partition` | incompatible DDL  | A DDL statement that exchanges a partition between two tables, such as the `ALTER TABLE EXCHANGE PARTITION` statement |
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**